### ISSUE TITLE
class inherit question

### ISSUE THREAD

**Issue by manaphp** [klesun/deep-assoc-completion#73]
great plugin!

```php
<?php

class Base
{

    /**
     * @return array
     */
    public static function sample()
    {
        return ['stub_id' => 1];
    }

    /**
     * @param array $filters
     * @param array $fields =static::sample()
     *
     * @return array
     */
    public static function find($filters, $fields = null)
    {
        return [];
    }
}

class Child extends Base
{
    public static function sample()
    {
        return ['user_id' => 1, 'user_name' => 'mark'];
    }
}

Child::find([], ['!!caret!!']);
```
!!caret!! complete only `stub_id` not `user_id` and `user_name`.

**Comment by manaphp** [comment]
`static` support is very important, when use `model`!!!

**Comment by klesun** [comment]
I'll take a look if this can be done without much blood... Resolving static would probabl cmplicate things a lot both code-wise and performance-wise.

**Comment by manaphp** [comment]
`model` needs many fields name auto completion!!

**Comment by klesun** [comment]
Related issue: #42

**Comment by klesun** [comment]
Added `static::` support in doc comments in [2019.04.27.001](https://plugins.jetbrains.com/plugin/9927-deep-assoc-completion/update/61505)
![image](https://user-images.githubusercontent.com/5202330/56854533-e60dd380-6940-11e9-95bf-f93dd693adb6.png)

**Comment by manaphp** [comment]
thanks,very good!!

**Comment by manaphp** [comment]
> 
> 
> Added `static::` support in doc comments in [2019.04.27.001](https://plugins.jetbrains.com/plugin/9927-deep-assoc-completion/update/61505)
> ![image](https://user-images.githubusercontent.com/5202330/56854533-e60dd380-6940-11e9-95bf-f93dd693adb6.png)

if `find` is not `static` method ,it does not work..

**Comment by klesun** [comment]
How does your phpdoc looks like when it's not static?

**Comment by klesun** [comment]
Oh, reproduced. I'll take a look.

**Comment by manaphp** [comment]
> 
> 
> How does your phpdoc look like when it's not static?
```php
  /**
     * @param string $field =array_keys(static::sample())[$i]
     *
     * @return bool
     */
    public function hasField($field)




$city =City::first([]);
$city->hasField('!caret!')

```

**Comment by klesun** [comment]
There is a temprary hackish workaround: if you specify it like this:
```
* @param string $field =array_keys($this->sample())[$i]
```
It will work in a non-static method

**Comment by klesun** [comment]
But I guess I'll make it possibly to use `static::` as well...

**Comment by manaphp** [comment]
> 
> 
> But I guess I'll make it possibly to use `static::` as well...

consistency is very import..

**Comment by klesun** [comment]
Fixed in beta release: [2019.04.29.001](https://plugins.jetbrains.com/files/9927/61570/deep-assoc-completion.jar?updateId=61570&pluginId=9927&family=intellij)

Will be in stable repo within a week.

**Comment by klesun** [comment]
Just uploaded to the main repo - https://plugins.jetbrains.com/plugin/9927-deep-assoc-completion/update/61899
You should now see 2019.04.29.001 in the list in your IDE updates
